packages/client/lib/net/peer/rlpxpeer.ts

```diff
@@ -112,27 +112,26 @@ export class RlpxPeer extends Peer {
       tcpPort: this.port,
     })
     this.rlpx.on('peer:error', (rlpxPeer: Devp2pRlpxPeer, error: Error) => {
-      this.emit('error', error)
+      this.config.events.emit(Event.PEER_ERROR, error, '')
     })
     this.rlpx.once('peer:added', async (rlpxPeer: Devp2pRlpxPeer) => {
       try {
         await this.bindProtocols(rlpxPeer)
-        this.emit('connected')
+        this.config.events.emit(Event.PEER_CONNECTED, rlpxPeer)
       } catch (error) {
-        this.emit('error', error)
+        this.config.events.emit(Event.PEER_ERROR, error, '')
```

Comment:
what's this third param supposed to be? the peer ID? I'm seeing a lot of empty strings

Reply:
I've re-worked this slightly and revised the `PEER_ERROR` event so it passed the Error object and then the `Peer` object.  The original string there was an artifact of legacy code that I hadn't properly dealt with before.  I've updated all the existing uses of this event to reflect this new definition.